backend/engine/plugins/gitlab_repo_health/rules/first_order_rules/branch_protection_pull_requests.py

```diff
@@ -0,0 +1,81 @@
+from requests import HTTPError
+
+from engine.plugins.gitlab_repo_health.utilities.gitlab import Gitlab
+from ..helpers import add_metadata, is_subdict_of, severity_schema
+
+
+class BranchProtectionPullRequests:
+    identifier = "branch_protection_pull_requests"
+    name = "Branch Protection - Require Pull Requests"
+    description = "Requires that a branch protection rule is enabled that requires pull requests"
+
+    config_schema = {
+        "type": "object",
+        "additionalProperties": False,
+        "properties": {
+            "type": {"const": identifier},
+            "id": {"type": "string"},
+            "enabled": {"type": "boolean"},
+            "name": {"type": "string"},
+            "description": {"type": "string"},
+            "severity": severity_schema,
+            "expect": {"type": "object"},
+            "min_approvals": {"type": "number"},
+        },
+    }
+
+    @staticmethod
+    def check(gitlab: Gitlab, owner: str, repo: str, branch: str, config={}):
+        min_approvals = config.get("min_approvals", 0)
+        requirements = config.get("expect", {})
+        approval_rules = []
+        approvals_config = None
+        try:
+            protection_config = gitlab.get_branch_protection(owner, repo, branch)
+            if requirements:
+                approvals_config = gitlab.get_approvals(owner, repo)
+
+            if min_approvals:
+                approval_rules = gitlab.get_approval_rules(owner, repo)
+
+        except HTTPError as e:
+            return add_metadata(
+                False,
+                BranchProtectionPullRequests,
+                config,
+                error_message=str(e),
+            )
+
+        push_levels = protection_config.get("push_access_levels", [])
+
+        if len(push_levels) != 1:
```

Comment:
What is `push_levels` here? Is there ever a usecase to have 0 or 2+ of them? If so, IMO, this should be a config item for the rule

Reply:
Ah, that makes sense. Reading the code, that wasn't super intuitive. What do you think about factoring out some of this logic into a `check_no_push_permissions` function (or similar) so that the code is a little more self-documenting?

If that's something you agree with, the same could be done for the `approvals_config`/`requirements_result` logic and the `min_approvals` logic.